src/hooks/screen/useAnonymousChatScreenHook.ts

```diff
@@ -0,0 +1,99 @@
+// eslint-disable-next-line import/no-unresolved
+import 'react-native-get-random-values';
+
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import UseAnonymousChatScreenHook from '../../../types/hooks/screens/useAnonymousChatScreenHook.types';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {getAnonymousUserId} from '../../utils/users';
+import {getUserId} from '../../utils/token';
+import {randomString} from '../../utils/string/StringUtils';
+
+function useAnonymousChatScreenHook(): UseAnonymousChatScreenHook {
+  const {localDb, chat, refresh} = useLocalDatabaseHook();
+  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen} = useChatUtilsHook();
+
+  const [chats, setChats] = React.useState<ChatSchema[]>([]);
+
+  const initChatData = async () => {
+    if (!localDb && !selectedChannel) return;
+    const myUserId = await getUserId();
+    const myAnonymousId = await getAnonymousUserId();
+    const data = (await ChatSchema.getAll(
+      localDb,
+      selectedChannel?.id,
+      myUserId,
+      myAnonymousId
+    )) as ChatSchema[];
+    setChats(data);
+  };
+
+  const sendChat = async (
+    message: string = randomString(20),
+    iteration = 0,
+    sendingChatSchema: ChatSchema = null
+  ) => {
+    if (iteration > 5) {
+      SimpleToast.show("Can't send message, please check your connection");
+      return;
+    }
+
+    let currentChatSchema = sendingChatSchema;
+
+    const myAnonymousId = await getAnonymousUserId();
+    try {
+      const randomId = uuid();
+
+      if (currentChatSchema === null) {
+        currentChatSchema = await ChatSchema.generateSendingChat(
+          randomId,
+          myAnonymousId,
+          selectedChannel?.id,
+          message,
+          localDb
+        );
+
+        await currentChatSchema.save(localDb);
+        refresh('chat');
+        refresh('channelList');
+      }
+
+      const response = await AnonymousMessageRepo.sendAnonymousMessage(
+        selectedChannel?.id,
+        message
+      );
+
+      await currentChatSchema.updateChatSentStatus(localDb, response);
+      refresh('chat');
+      refresh('channelList');
+    } catch (e) {
+      console.log(e);
+      if (e?.response?.data?.status === 'Channel is blocked') return;
+
+      setTimeout(() => {
+        sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
+          console.log(sendChatError);
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling could be improved here. Currently, it's just logging the error and not providing any feedback to the user. Consider showing a toast message or some other form of user-friendly error message. Also, it's generally not a good practice to ignore errors silently. If 'Channel is blocked' is an expected error, handle it explicitly.
```diff
-      console.log(e);
-      if (e?.response?.data?.status === 'Channel is blocked') return;
-
-      setTimeout(() => {
-        sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
-          console.log(sendChatError);
-        });
-      }, 1000);
+      if (e?.response?.data?.status === 'Channel is blocked') {
+        SimpleToast.show(ERROR_MESSAGES.CHANNEL_BLOCKED);
+      } else {
+        SimpleToast.show(ERROR_MESSAGES.SEND_CHAT_ERROR);
+        setTimeout(() => {
+          sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
+            console.error(sendChatError);
+          });
+        }, 1000);
+      }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->